mongodbatlas/resource_mongodbatlas_cluster_test.go

```diff
@@ -12,51 +12,226 @@ import (
+						Check: resource.ComposeTestCheckFunc(
+							testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+							testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+							resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+							resource.TestCheckResourceAttr(resourceName, "name", name),
+							resource.TestCheckResourceAttrSet(resourceName, "mongo_uri"),
+							resource.TestCheckResourceAttrSet(resourceName, "replication_specs.#"),
+							resource.TestCheckResourceAttrSet(resourceName, "replication_specs.0.regions_config.#"),
+						),
+					},
+				},
+			})
+		})
+	}
 
 }
+func TestAccResourceMongoDBAtlasCluster_basicGCP(t *testing.T) {
+	var cluster matlas.Cluster
+
+	resourceName := "mongodbatlas_cluster.test"
+	projectID := os.Getenv("MONGODB_ATLAS_PROJECT_ID")
+	var region string
+
+	for tier, size := range matlas.DefaultDiskSizeGB["GCP"] {
+		name := fmt.Sprintf("testAcc-%s-%s-%ggb-%s", "GCP", tier, size, acctest.RandString(1))
+
+		switch tier {
+		case "M200":
+			region = "EUROPE_WEST_3"
+		case "M300":
+			region = "CENTRAL_US"
+		default:
+			region = "US_EAST_4"
+		}
+
+		t.Run(name, func(t *testing.T) {
+			resource.ParallelTest(t, resource.TestCase{
+				PreCheck:     func() { testAccPreCheck(t) },
+				Providers:    testAccProviders,
+				CheckDestroy: testAccCheckMongoDBAtlasClusterDestroy,
+				Steps: []resource.TestStep{
+					{
+						Config: testAccMongoDBAtlasClusterConfigGCP(projectID, name, "true", tier, region),
+						Check: resource.ComposeTestCheckFunc(
+							testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+							testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+							resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+							resource.TestCheckResourceAttr(resourceName, "name", name),
+							resource.TestCheckResourceAttrSet(resourceName, "mongo_uri"),
+							resource.TestCheckResourceAttrSet(resourceName, "replication_specs.#"),
+							resource.TestCheckResourceAttrSet(resourceName, "replication_specs.0.regions_config.#"),
+						),
+					},
+					{
+						Config: testAccMongoDBAtlasClusterConfigGCP(projectID, name, "false", tier, region),
+						Check: resource.ComposeTestCheckFunc(
+							testAccCheckMongoDBAtlasClusterExists(resourceName, &cluster),
+							testAccCheckMongoDBAtlasClusterAttributes(&cluster, name),
+							resource.TestCheckResourceAttrSet(resourceName, "project_id"),
+							resource.TestCheckResourceAttr(resourceName, "name", name),
+							resource.TestCheckResourceAttrSet(resourceName, "mongo_uri"),
+							resource.TestCheckResourceAttrSet(resourceName, "replication_specs.#"),
+							resource.TestCheckResourceAttrSet(resourceName, "replication_specs.0.regions_config.#"),
+						),
+					},
+				},
+			})
+		})
+	}
+}
+
+func TestAccResourceMongoDBAtlasCluster_tenant(t *testing.T) {
+	var cluster matlas.Cluster
+
+	resourceName := "mongodbatlas_cluster.tenant"
```

Comment:
this might be confusing but while we use the word Tenant in the API this is what we refer to as our "shared" tier.  Might make more sense to use shared instead of tenant for readability.

Reply:
Did you see this @PacoDw ? Any objections to calling it Shared?

Reply:
No, everything is ok for me 😅. Sorry, I was flying yesterday